src/desktop/config/ConfigFile.ts

```diff
@@ -9,34 +10,37 @@ type FsExports = typeof FsModule
  * @param p {string} path to file the configFile should write to
  * @param fs the fs object returned by "import fs from ''fs"
  */
-export function getConfigFile(p: string, fs: FsExports): ConfigFile {
-	if (!Object.keys(instances).includes(p)) {
-		instances[p] = new ConfigFile(p, fs)
+export function getConfigFile(p: string, f: string, fs: FsExports): ConfigFile {
+	const fullpath = path.join(p, f)
+	if (!Object.keys(instances).includes(fullpath)) {
+		instances[fullpath] = new ConfigFile(p, f, fs)
 	}
 
-	return instances[p]
+	return instances[fullpath]
 }
 
 export type ConfigFileType = ConfigFile
 
-class ConfigFile {
-	_path: string
-	_accessPromise: Promise<any>
-	readonly _fs: FsExports
+export class ConfigFile {
+	private path: string
+	private fullpath: string
+	private accessPromise: Promise<any>
+	private readonly fs: FsExports
 
 	/**
 	 * @param p path to the file the json objects should be stored in
 	 * @param fs
 	 **/
-	constructor(p: string, fs: FsExports) {
-		this._path = p
-		this._accessPromise = Promise.resolve()
-		this._fs = fs
+	constructor(p: string, f: string, fs: FsExports) {
+		this.path = p
```

Comment:
path, fullpath and fs could be declared as `private readonly` inside of the constructor signature declaration like you did in `DesktopConfig`